gifmodify.js

```diff
@@ -157,14 +158,76 @@ function addColorShiftingFramesGIF(inputGif, options, callback) {
 
   for (let i = 0; i < frames.length; i++) {
     let frame = frames[i];
-    setFrameProperties(frame);
+    setFrameProperties(frame, { interlaced: false });
     shiftColors(frame.bitmap, interval * i / frames.length, randomBlack, randomWhite);
   }
   callback(frames);
 }
 
-function setFrameProperties(frame) {
-  frame.interlaced = false;
+exports.createWigglingGIF = function(options) {
+  return new Promise((resolve, reject) => {
+    getGifFromBuffer(options.buffer).then(inputGif => {
+      addWigglingFramesGIF(inputGif, options, frames => {
+        let codec = new GifCodec();
+        codec.encodeGif(frames).then(resultGif => {
+          resolve(resultGif.buffer);
+        });
+      });
+    }).catch(error => reject(error));
+  });
+}
+
+function addWigglingFramesGIF(inputGif, options, callback) {
```

Comment:
This function is very similar to addWigglingFramesPNG, can you put this in a separate function like I did with rainbow?